sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/Window.java

```diff
@@ -332,7 +332,7 @@
   public WindowingStrategy<?, ?> getOutputStrategyInternal(WindowingStrategy<?, ?> inputStrategy) {
     WindowingStrategy<?, ?> result = inputStrategy;
     if (getWindowFn() != null) {
-      result = result.withWindowFn(getWindowFn());
+      result = result.withAlreadyMerged(false).withWindowFn(getWindowFn());
```

Comment:
I'm not sure if we can safely assume getWindowFn() is always non-null. Wouldn't it be safer to set already merged directly like WindowingStrategy<?, ?> result = inputStrategy.withAlreadyMerged(false)?

Reply:
This is a "model" question, sort of:

 - if the user calls `Window.into(WindoFn)` we definitely reset `alreadyMerged`.
 - if the user calls `Window.remerge()` we reset `alreadyMerged`.
 - if the user calls `Window.configure()` to set a new trigger, accumulation mode, or allowed lateness: should we reset `alreadyMerged`? I decided that we should not. So if windowFn is null, we do not reset.

Reply:
I see, thanks for the explanation.